homeassistant/components/lyric/__init__.py

```diff
@@ -0,0 +1,186 @@
+"""
+Support for Honeywell Lyric devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/lyric
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import CONF_SCAN_INTERVAL, HTTP_BAD_REQUEST
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+_CONFIGURING = {}
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['http']
+
+REQUIREMENTS = ['python-lyric==1.1.4']
+
+DOMAIN = 'lyric'
+
+DATA_LYRIC = 'lyric'
+
+LYRIC_CONFIG_FILE = 'lyric.conf'
+CONF_CLIENT_ID = 'client_id'
+CONF_CLIENT_SECRET = 'client_secret'
+CONF_REDIRECT_URI = 'redirect_uri'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_CLIENT_ID): cv.string,
+        vol.Required(CONF_CLIENT_SECRET): cv.string,
+        vol.Optional(CONF_REDIRECT_URI): cv.string,
+        vol.Optional(CONF_SCAN_INTERVAL, default=270): cv.positive_int
+    })
+}, extra=vol.ALLOW_EXTRA)
+
+
+def request_configuration(lyric, hass, config):
```

Comment:
We shouldn't add new integrations using the configurator. Please use a config flow with an external step instead.

Reply:
See the somfy integration for an example of a config flow like that.

Reply:
Thanks for the suggestion! Decided to rewrite the integration so it doesn't use the config and instead directly uses config flow. Much more easier to configure now just using the client id and secret then sending the user to honeywell to authenticate